### PR TITLE
[Merged by Bors] - fix(*): revert tactic state propagation through conv

### PR DESCRIPTION
**Pull request by eric-wieser**
This turned out to break some quite simple cases

https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/conv.20in.20lean.20master/near/214359311

### CODE REVIEW

File: tests/lean/conv_for_find.lean [hunk 1]
```diff
@@ -29,6 +29,7 @@ example {a b : ℕ} : ∃ x, x * a = a * x :=  begin
   },
 end
 
+-- but this unfortunately doesn't work for `find` and `for`
```
Comment (eric-wieser) [inline tests/lean/conv_for_find.lean:32]: Once CI fails I'll update the error comparison file with the errors these tests produce.

**Comment by gebner** [comment]
Could you please add a test case?

File: tests/lean/conv_for_find.lean.expected.out [hunk 2]
```diff
@@ -10,3 +10,11 @@ conv_for_find.lean:19:6: error: unknown identifier 'does_not_exist'
...
 a b c : ℕ
 ⊢ a * b * c = ?m_1
+conv_for_find.lean:40:0: error: tactic failed, there are unsolved goals
+state:
+a b : ℕ
+⊢ ℕ
+conv_for_find.lean:49:0: error: tactic failed, there are unsolved goals
+state:
+a b : ℕ
+⊢ ℕ
```
Comment (eric-wieser) [inline tests/lean/conv_for_find.lean.expected.out:20]:
```suggestion
⊢ ℕ
p i
p i
p j
```

**Comment by eric-wieser** [comment]
Test added. Can you squash the commits when you merge through the github UI? I've been editing through the web UI because I haven't attempted to get a local build system set up.

**Comment by gebner** [comment]
I'll leave the squashing to:
bors r+

**Pull request closed without merging**